UnityProject/Assets/Scripts/Objects/Machines/Griddle.cs

```diff
@@ -185,21 +187,29 @@ private void CheckCooked(float cookTime)
 
 		private void OnSyncPlayAudioLoop(bool oldState, bool newState)
 		{
+
+			// Only one thread should be able to try to play the sound at a time.
+			// Otherwise causes issues where while one thread is waiting another thread
+			// can get in here to request it a second time and the sound plays twice.
+			audioLoopLock.WaitOne();
```

Comment:
I'm slightly confused. To my knowledge none of this is threaded. 

@Bod9001 You didn't thread Mirror for this sort of stuff right?

Reply:
I'm not certain but I think it is something built into Unity with the `[SyncVar(hook = nameof(OnSyncPlayAudioLoop))]` on line 55 above. Their documentation on it is sparse and basically just says it exists so it is not very helpful.

Reply:
> I'm not certain but I think it is something built into Unity with the `[SyncVar(hook = nameof(OnSyncPlayAudioLoop))]` on line 55 above. Their documentation on it is sparse and basically just says it exists so it is not very helpful.

It's not from Unity. It's our networking package Mirror.

https://mirror-networking.gitbook.io/docs/guides/synchronization/syncvar-hooks

Reply:
could be further is receiving synchvar updates , but I doubt it